core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -256,8 +271,7 @@ class Partition(val topicPartition: TopicPartition,
     tags
   )
 
-  def isUnderReplicated: Boolean =
-    isLeader && inSyncReplicaIds.size < allReplicaIds.size
+  def isUnderReplicated: Boolean = (assignmentState.replicationFactor - inSyncReplicaIds.size) > 0
```

Comment:
Missing `isLeader` check I believe

Reply:
Thanks, it did disappear indeed. Will fix.